src/raft_baselines/classifiers/transformers_causal_lm_classifier.py

```diff
@@ -0,0 +1,78 @@
+from logging import Formatter
+from typing import Dict, Optional, List, Mapping
+
+import numpy as np
+import datasets
+import torch
+from transformers import AutoModelForCausalLM
+from sentence_transformers import util
+
+from raft_baselines.classifiers.in_context_classifier import InContextClassifier
+from raft_baselines.utils.tokenizers import TransformersTokenizer
+from raft_baselines.utils.embedders import SentenceTransformersEmbedder
+
+
+class TransformersCausalLMClassifier(InContextClassifier):
+    def __init__(
+        self,
+        *args,
+        model_type: str = "distilgpt2",
+        **kwargs,
+    ) -> None:
+        tokenizer = TransformersTokenizer(model_type)
+        self.model = AutoModelForCausalLM.from_pretrained(model_type)
+        self.similarity_embedder = SentenceTransformersEmbedder()
+
+        super().__init__(
+            *args,
+            tokenizer=tokenizer,
+            max_tokens=self.model.config.max_position_embeddings,
+            **kwargs,
+        )
+
+    def semantically_select_training_examples(
```

Comment:
Just an idea, but if you didn't want to introduce the `sentence-transformers` dependency, you can do semantic search directly in `datasets`: https://huggingface.co/docs/datasets/faiss_and_ea.html#adding-a-faiss-index

Reply:
I'll leave as is for now

Reply:
sounds good!